distribution/test/itests/test-itests-alliance/src/test/java/org/codice/alliance/test/itests/NsiliEndpointTest.java

```diff
@@ -0,0 +1,242 @@
+        assertThat(sampleNsiliClient.get_related_files(result), is(new String[]{expectedThumbnailFilename}));
+        assertStubServerPutRequest(expectedThumbnailFilename, "image/jpeg");
+
+        // OrderMgr
+        String orderResultFilename = sampleNsiliClient.testOrder(result);
+        // TODO test processing packageElements of the order response https://codice.atlassian.net/browse/CAL-192
+        assertStubServerPutRequest(orderResultFilename, "application/x-tar");
+
+        deleteMetacard(ingestedProductId);
+        sampleNsiliClient.cleanup();
+    }
+
+    @Test
+    public void testCatalogMgrViaCallback() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        sampleNsiliClient.testCallbackCatalogMgr();
+
+        sampleNsiliClient.cleanup();
+    }
+
+    private void startHttpListener() {
+        server = new StubServer(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort())).run();
+        whenHttp(server).match(method(Method.PUT), Condition.startsWithUri(NSILI_FILE_URI_PATH))
+                .then(Action.success());
+    }
+
+    private void assertStubServerPutRequest(String expectedFilename, String expectedType) {
+        assertThat(server.getCalls().size(), is(1));
+        final Call call = server.getCalls().get(0);
+        assertThat(call.getMethod(), is(Method.PUT));
+        assertThat(call.getUrl(), is(RESTITO_STUB_SERVER.getUrl() + "/" + expectedFilename));
+        assertThat(call.getContentType(), is(expectedType));
+        server.clearCalls();
+    }
+
+    private String ingestRecord(String fileName, String fileType)
+            throws IOException, InterruptedException {
+        InputStream inputStream = getClass().getClassLoader()
+                .getResourceAsStream(fileName);
+        byte[] fileBytes = IOUtils.toByteArray(inputStream);
+
+        getServiceManager().waitForSourcesToBeAvailable(REST_PATH.getUrl());
+
+        return given().multiPart("file", fileName, fileBytes, fileType)
+                .expect()
+                .statusCode(HttpStatus.SC_CREATED)
+                .when()
+                .post(REST_PATH.getUrl())
+                .getHeader("id");
+    }
+
+    private void deleteMetacard(String id) {
+        delete(REST_PATH.getUrl() + id);
+    }
+
+    private void configureSecurityStsClient() throws IOException, InterruptedException {
+        Configuration stsClientConfig = configAdmin.getConfiguration(
+                "ddf.security.sts.client.configuration.cfg",
+                null);
+        Dictionary<String, Object> properties = new Hashtable<>();
+
+        properties.put("address",
+                DynamicUrl.SECURE_ROOT + HTTPS_PORT.getPort()
+                        + "/services/SecurityTokenService?wsdl");
+        stsClientConfig.update(properties);
+    }
+
+    public class NsiliSourceProperties extends HashMap<String, Object> {
```

Comment:
We're trying to move away from extending `HashMap` in itests and instead have a private method that builds and returns a `Map` with the right values in it.